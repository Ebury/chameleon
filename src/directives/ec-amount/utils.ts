
import defaultOptions from './options';
import type { AmountDirectiveOptions } from './types';

function format(input: undefined, opts?: AmountDirectiveOptions): undefined;
function format(input: null, opts?: AmountDirectiveOptions): null;
function format(input: string | number, opts?: AmountDirectiveOptions): string;
function format(input: string | number | undefined | null, opts: AmountDirectiveOptions = defaultOptions): string | null | undefined {
  /* eslint-disable no-param-reassign */
  if (input === '' || input === null || input === undefined) {
    return input;
  }

  if (typeof input === 'number') {
    input = input.toString();
  }

  const negative = input.startsWith('-') ? '-' : '';

  input = sanitizeInput(input, opts.decimalSeparator);
  if (opts.precision > 0) {
    input = keepOnlyOneDecimalSeparator(input, opts.decimalSeparator);
  } else {
    input = removeDecimal(input, opts.decimalSeparator);
  }
  const parts = input.split(opts.decimalSeparator);
  let integralPart = parseToStr(parts[0]);
  let fractionalPart = parts[1];

  if (fractionalPart && fractionalPart.length > opts.precision) {
    fractionalPart = fractionalPart.substring(0, opts.precision);
  }

  integralPart = addGroupingSeparators(integralPart, opts.groupingSeparator);
  const result = negative + joinIntegralAndFractionalParts(integralPart, fractionalPart, opts.decimalSeparator);
  return result;
}

function parseToStr(number: string): string {
  const parsed = Number.parseFloat(number);
  if (Number.isNaN(parsed)) {
    return '';
  }
  return parsed.toString();
}

function keepOnlyOneDecimalSeparator(number: string, decimalSeparator: string): string {
  if (number.includes(decimalSeparator)) {
    const split = number.split(decimalSeparator);
    return split[0] + decimalSeparator + split.slice(1).join('');
  }
  return number;
}

function removeDecimal(number: string, decimalSeparator: string): string {
  const split = number.split(decimalSeparator);
  return split[0];
}

function unFormat(input: string, groupingSeparator: string, decimalSeparator: string): string {
  if (!input) {
    return input;
  }

  return input
    .replace(new RegExp(`\\${groupingSeparator}`, 'g'), '')
    .replace(new RegExp(`\\${decimalSeparator}`, 'g'), '.');
}

function sanitizeInput(input: string, decimalSeparator: string): string {
  return input.replace(new RegExp(`[^0-9${decimalSeparator}]`, 'g'), '') || '';
}

function addGroupingSeparators(integralPart: string, separator: string): string {
  return integralPart.replace(/(\d)(?=(?:\d{3})+\b)/gm, `$1${separator}`);
}

function joinIntegralAndFractionalParts(integralPart: string, fractionalPart: string, decimalSeparator: string): string {
  return fractionalPart !== undefined ? integralPart + decimalSeparator + fractionalPart : integralPart;
}

/* c8 ignore start */
function setCursor(el: HTMLInputElement, position: number) {
  const setSelectionRange = function setSelectionRange() {
    el.setSelectionRange(position, position);
  };

  if (el === document.activeElement) {
    setSelectionRange();
    setTimeout(setSelectionRange, 1); // Android Fix
  }
}
/* c8 ignore stop */

export {
  format,
  sanitizeInput,
  setCursor,
  unFormat,
};
