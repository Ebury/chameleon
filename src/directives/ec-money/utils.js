import defaults from './options';

function format(input, opt = defaults) {
  if (input === '' || input === null || input === undefined) {
    return input;
  }

  if (typeof input === 'number') {
    input = input.toString();
  }

  const negative = input.includes('-') ? '-' : '';

  input = sanitizeInput(input, opt.decimalSeparator);
  input = keepOnlyOneDecimalSeparator(input, opt.decimalSeparator);
  const parts = input.split(opt.decimalSeparator);
  let integer = parseToStr(parts[0]);
  let decimal = parts[1];

  if (decimal && decimal.length > opt.precision) {
    decimal = decimal.substr(0, opt.precision);
  }

  integer = addGroupingSeparators(integer, opt.groupingSeparator);
  const result = negative + joinIntegerAndDecimal(integer, decimal, opt.decimalSeparator);
  return result;
}

function parseToStr(number) {
  /* eslint no-restricted-globals: "off" */
  const parsed = parseFloat(number);
  if (isNaN(parsed)) {
    return '';
  }
  return parsed.toString();
}

function keepOnlyOneDecimalSeparator(number, decimalSeparator) {
  if (number.includes(decimalSeparator)) {
    const split = number.split(decimalSeparator);
    return split[0] + decimalSeparator + split.slice(1).join('');
  }
  return number;
}

function unFormat(input, groupingSeparator, decimalSeparator) {
  /* eslint no-param-reassign: "off" */
  input = (`${input}`).replace(new RegExp(`\\${groupingSeparator}`, 'g'), '');
  input = (`${input}`).replace(new RegExp(`\\${decimalSeparator}`, 'g'), '.');
  return input;
}

function sanitizeInput(input, decimalSeparator) {
  return input.replace(new RegExp(`[^0-9${decimalSeparator}]`, 'g'), '') || '';
}

function addGroupingSeparators(integer, separator) {
  return integer.replace(/(\d)(?=(?:\d{3})+\b)/gm, `$1${separator}`);
}

function joinIntegerAndDecimal(integer, decimal, decimalSeparator) {
  return decimal !== undefined ? integer + decimalSeparator + decimal : integer;
}

function setCursor(el, position) {
  const setSelectionRange = function setSelectionRange() {
    el.setSelectionRange(position, position);
  };

  if (el === document.activeElement) {
    setSelectionRange();
    setTimeout(setSelectionRange, 1); // Android Fix
  }
}

// https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events#The_old-fashioned_way
function event(name) {
  const evt = document.createEvent('Event');
  evt.initEvent(name, true, true);
  return evt;
}

export {
  format,
  unFormat,
  setCursor,
  event,
  sanitizeInput,
};
