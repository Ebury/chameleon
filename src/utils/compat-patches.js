/* istanbul ignore file */

import {
  Dropdown as FvDropdown,
  TooltipDirective as FvTooltipDirective,
  VClosePopper,
  VTooltip,
} from 'floating-vue';

// we need to tell the @vue/compat that all components coming from floating-vue are Vue 3 components so it skips
// them when patching component instances. These components can't run in default mode for Vue 2, there is lots of
// errors coming from the console in that mode.
export default function applyCompatPatches() {
  const compatConfig = {
    MODE: 3,
    RENDER_FUNCTION: false,
  };

  VTooltip.compatConfig = compatConfig;
  VClosePopper.compatConfig = compatConfig;
  FvTooltipDirective.compatConfig = compatConfig;
  for (const component of Object.values(FvTooltipDirective.components)) {
    component.compatConfig = compatConfig;
  }

  FvDropdown.compatConfig = compatConfig;

  for (const component of Object.values(FvDropdown.components)) {
    component.compatConfig = compatConfig;
  }
}
