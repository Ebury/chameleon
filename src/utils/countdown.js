import mitt from 'mitt';

export default class Countdown {
  constructor() {
    const m = mitt();
    Object.assign(this, m);
  }

  reduceMinutesLeft(totalSeconds) {
    this.getminutes(totalSeconds);
    this.getseconds(totalSeconds, this.minutesLeft);
  }

  start(secondsToGo, showMinutes) {
    this.seconds = secondsToGo;
    this.secondsLeft = secondsToGo;
    this.currentTime = Math.ceil(Date.now() / 1000);
    this.startTime = Math.ceil(Date.now() / 1000);
    this.interval = setInterval(() => this.reduceSecondsLeft(showMinutes), 1000);
  }

  get timeDifference() {
    return this.currentTime - this.startTime;
  }

  reduceSecondsLeft(showMinutes) {
    this.currentTime = Math.ceil(Date.now() / 1000);
    this.totalSeconds = this.seconds - this.timeDifference;
    this.minutesLeft = null;
    this.secondsLeft = this.totalSeconds;

    if (showMinutes) {
      this.reduceMinutesLeft(this.totalSeconds);
    }

    this.emit('time-updated', {
      seconds: this.secondsLeft,
      minutes: this.minutesLeft,
      totalSeconds: this.totalSeconds,
    });

    if (this.totalSeconds <= 0) {
      this.stop();
      this.emit('time-expired');
    }
  }

  stop() {
    clearInterval(this.interval);
  }

  getminutes(totalSeconds) {
    this.minutesLeft = Math.floor(totalSeconds / 60);
  }

  getseconds(totalSeconds, minutes) {
    // take mins remaining (as seconds) away from total seconds remaining
    this.secondsLeft = totalSeconds - Math.round(minutes * 60);
  }
}
