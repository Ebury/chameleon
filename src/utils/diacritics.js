export function removeDiacritics(str) {
  if (typeof str === 'string' && str.normalize) {
    // https://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript
    return str.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
  }

  // IE11 users will not enjoy this feature:
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
  return str;
}
