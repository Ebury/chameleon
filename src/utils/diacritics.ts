export function removeDiacritics(str: string): string {
  if (str && str.normalize) {
    // https://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript
    return str.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
  }

  // IE11 users will not enjoy this feature:
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
  return str;
}
