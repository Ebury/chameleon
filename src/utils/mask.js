export function mask(text, maskSymbol = '*', visibleChars = 2) {
  if (!text) {
    throw new Error('Text is required');
  }
  if (!maskSymbol) {
    throw new Error('Mask symbol cannot be empty');
  }
  if (!visibleChars || visibleChars < 0) {
    throw new Error('Visible characters must be a number greater than zero');
  }

  const digitsToBeMasked = text.length - visibleChars;
  return maskSymbol.repeat(digitsToBeMasked) + text.substr(digitsToBeMasked);
}
