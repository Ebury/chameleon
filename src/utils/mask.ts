export function mask(text: string, maskSymbol: string = '*', visibleChars: number = 2): string {
  if (!text) {
    throw new Error('Text is required');
  }
  if (!maskSymbol) {
    throw new Error('Mask symbol cannot be empty');
  }
  if (!visibleChars || visibleChars < 0) {
    throw new Error('Visible characters must be a number greater than zero');
  }

  const digitsToBeMasked = text.length - visibleChars;
  return maskSymbol.repeat(digitsToBeMasked) + text.substring(digitsToBeMasked);
}
