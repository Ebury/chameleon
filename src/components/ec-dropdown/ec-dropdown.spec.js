import Vue from 'vue';
import { mount, createLocalVue } from '@vue/test-utils';
import EcDropdown from './ec-dropdown.vue';
import { withMockedConsole } from '../../../tests/utils/console';

describe('EcDropdown', () => {
  const MockedEcPopover = Vue.extend({
    methods: {
      update: jest.fn(),
    },
    template: '<div data-popover-stub><slot /><slot name="popover" /></div>',
  });

  function mountDropdown(props, mountOpts) {
    const localVue = createLocalVue();

    return mount(EcDropdown, {
      localVue,
      propsData: { ...props },
      stubs: { EcPopover: MockedEcPopover },
      ...mountOpts,
    });
  }

  function mountAsTemplate(template, props, mountOpts) {
    const localVue = createLocalVue();

    const Component = localVue.extend({
      components: { EcDropdown },
      template,
    });

    return mount(Component, {
      localVue,
      propsData: { ...props },
      stubs: { EcPopover: MockedEcPopover },
      ...mountOpts,
    });
  }

  const items = [
    { id: 1, text: 'Item 1' },
    { id: 2, text: 'Item 2' },
    { id: 3, text: 'Item 3' },
    {
      id: 4, text: 'Item 4', disabled: true, disabledReason: 'Random text',
    },
    { id: 5, text: 'Item 5' },
  ];

  it('should render as expected', () => {
    const wrapper = mountDropdown();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render readonly input as a trigger', () => {
    const wrapper = mountDropdown();
    expect(wrapper.findByDataTest('ec-input-field__input').element).toMatchSnapshot();
  });

  it('should not enable searching by default', () => {
    const wrapper = mountDropdown();
    expect(wrapper.findByDataTest('ec-dropdown-search__search-area').exists()).toBe(false);
  });

  describe(':props', () => {
    it('should not render any label if prop is not given', () => {
      const wrapper = mountDropdown({ label: '' });
      expect(wrapper.findByDataTest('ec-input-field__label').exists()).toBe(false);
    });

    it.each([
      ['modal', false],
      ['tooltip', false],
      ['notification', false],
      ['level-1', false],
      ['level-2', false],
      ['level-3', false],
      ['random', true],
    ])('should validate if the level prop("%s") is on the allowed array of strings', (str, error) => {
      if (error) {
        withMockedConsole((errorSpy) => {
          mountDropdown({ items, level: str });
          expect(errorSpy).toHaveBeenCalledTimes(2);
          // this is the test for the dropdown
          expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: custom validator check failed for prop "level"');
          // this is the test for the
          expect(errorSpy.mock.calls[1][0]).toContain('Invalid prop: custom validator check failed for prop "level"');
        });
      } else {
        const wrapper = mountDropdown({ items, level: str });
        expect(wrapper.findByDataTest('ec-popover-dropdown-search').attributes('level')).toBe(str);
      }
    });

    it('should pass label prop to the triggering input', () => {
      const wrapper = mountDropdown({ label: 'Random label' });
      expect(wrapper.findByDataTest('ec-input-field__label').element).toMatchSnapshot();
    });

    it('should not render any error message if prop is not given', () => {
      const wrapper = mountDropdown({ errorMessage: '' });
      expect(wrapper.findByDataTest('ec-input-field__error-text').exists()).toBe(false);
    });

    it('should pass errorMessage prop to the triggering input', () => {
      const wrapper = mountDropdown({ error: 'Random error' });
      expect(wrapper.findByDataTest('ec-input-field__error-text').element).toMatchSnapshot();
    });

    it('should enable search when isSearchEnabled is set to true', () => {
      const wrapper = mountDropdown({ isSearchEnabled: true });
      expect(wrapper.findByDataTest('ec-dropdown-search__search-area').element).toMatchSnapshot();
    });

    it('should use placeholder prop', () => {
      const wrapper = mountDropdown({ placeholder: 'Random placeholder' });
      expect(wrapper.findByDataTest('ec-input-field__input').element).toMatchSnapshot();
    });

    it('should use searchPlaceholder prop if search is enabled', () => {
      const wrapper = mountDropdown({ searchPlaceholder: 'Random search placeholder', isSearchEnabled: true });
      expect(wrapper.findByDataTest('ec-dropdown-search__search-area').element).toMatchSnapshot();
    });

    it('should get disabled when disabled prop is set', () => {
      const wrapper = mountDropdown({ disabled: true });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should not show the items when the loading is set to true', () => {
      const wrapper = mountDropdown({ items, isLoading: true });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('#slots', () => {
    it('should forward item slot', () => {
      const wrapper = mountDropdown({
        items,
        selected: items[2],
      }, {
        scopedSlots: {
          item: '<div>({{ props.index }}) {{ props.item.text }} - {{ props.isSelected }}</div>',
        },
      });
      expect(wrapper.findByDataTest('ec-dropdown-search__item-list').element).toMatchSnapshot();
    });

    it('should forward CTA slot', () => {
      const wrapper = mountDropdown({}, {
        scopedSlots: {
          cta: '<button>My CTA</button>',
        },
      });
      expect(wrapper.findByDataTest('ec-dropdown-search__cta-area').element).toMatchSnapshot();
    });
  });

  describe('v-model', () => {
    it('should use the v-model', () => {
      const wrapper = mountAsTemplate(
        '<ec-dropdown :items="items" v-model="selected" />',
        {},
        {
          data() {
            return { items, selected: null };
          },
        },
      );

      expect(wrapper.vm.selected).toBe(null);
      selectItem(wrapper, 0);
      expect(wrapper.vm.selected).toBe(items[0]);
      selectItem(wrapper, 1);
      expect(wrapper.vm.selected).toBe(items[1]);
    });

    it('should display selected text in the readonly input', () => {
      const wrapper = mountDropdown({ items, selected: items[0] });
      expect(wrapper.findByDataTest('ec-input-field__input').element.value).toBe(items[0].text);
    });

    it('should display selectedText instead of selected item text if the prop is given', () => {
      const wrapper = mountDropdown({ items, selected: items[0], selectedText: 'Random text' });
      expect(wrapper.findByDataTest('ec-input-field__input').element.value).toBe('Random text');
    });

    it('should display empty text in the readonly input when nothing is selected', () => {
      const wrapper = mountDropdown({ items, selected: null });
      expect(wrapper.findByDataTest('ec-input-field__input').element.value).toBe('');
    });
  });

  describe('@events', () => {
    it('should emit change event when an item is selected', () => {
      const wrapper = mountDropdown({ items });
      selectItem(wrapper, 1);

      expect(wrapper.emitted('change').length).toEqual(1);
      expect(wrapper.emitted('change')[0]).toEqual([items[1]]);
    });

    it('should emit open event when the input is clicked(click mock)', () => {
      const wrapper = mountDropdown({ items });

      wrapper.findByDataTest('ec-popover-dropdown-search').vm.$emit('show');
      expect(wrapper.emitted('open').length).toBe(1);
    });

    it('should not return focus back to readonly input if it already has it', () => {
      const wrapper = mountDropdown({ items });
      const focusSpy = jest.spyOn(wrapper.find({ ref: 'trigger' }).element, 'querySelector')
        .mockImplementation(() => true);
      selectItem(wrapper, 1);

      expect(focusSpy).toHaveBeenCalledTimes(1);
      expect(wrapper.emitted('change')[0]).toEqual([items[1]]);
      focusSpy.mockRestore();
    });
  });
});

function selectItem(wrapper, index) {
  wrapper.findByDataTest('ec-dropdown__input').trigger('mousedown');
  wrapper.findByDataTest('ec-dropdown__input').trigger('click');
  wrapper.findAll('.ec-dropdown-search__item').at(index).trigger('click');
}
