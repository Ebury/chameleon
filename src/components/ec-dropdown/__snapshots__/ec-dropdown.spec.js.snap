// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcDropdown #slots should forward CTA slot 1`] = `
<li
  class="ec-dropdown-search__cta-area"
  data-test="ec-dropdown-search__cta-area"
>
  <button>
    My CTA
  </button>
</li>
`;

exports[`EcDropdown #slots should forward item slot 1`] = `
<ul
  class="ec-dropdown-search__item-list"
  data-test="ec-dropdown-search__item-list"
>
  <!---->
   
  <!---->
   
  <li
    class="ec-dropdown-search__item"
    data-test="ec-dropdown-search__item ec-dropdown-search__item--0"
    title="Item 1"
  >
    <div>
      (0) Item 1 - false
    </div>
  </li>
  <li
    class="ec-dropdown-search__item"
    data-test="ec-dropdown-search__item ec-dropdown-search__item--1"
    title="Item 2"
  >
    <div>
      (1) Item 2 - false
    </div>
  </li>
  <li
    class="ec-dropdown-search__item ec-dropdown-search__item--is-selected"
    data-test="ec-dropdown-search__item ec-dropdown-search__item--2"
    title="Item 3"
  >
    <div>
      (2) Item 3 - true
    </div>
  </li>
  <li
    class="ec-dropdown-search__item ec-dropdown-search__item--is-disabled ec-has-tooltip"
    data-original-title="Item 4"
    data-test="ec-dropdown-search__item ec-dropdown-search__item--3"
  >
    <div>
      (3) Item 4 - false
    </div>
  </li>
  <li
    class="ec-dropdown-search__item"
    data-test="ec-dropdown-search__item ec-dropdown-search__item--4"
    title="Item 5"
  >
    <div>
      (4) Item 5 - false
    </div>
  </li>
</ul>
`;

exports[`EcDropdown :props should enable search when isSearchEnabled is set to true 1`] = `
<li
  class="ec-dropdown-search__search-area"
  data-test="ec-dropdown-search__search-area"
>
  <svg
    class="ec-dropdown-search__search-icon ec-icon"
  >
    <use
      xlink:href="#ec-simple-search"
    />
  </svg>
   
  <input
    autocomplete="off"
    class="ec-dropdown-search__search-input"
    data-test="ec-dropdown-search__search-input"
    placeholder="Search..."
  />
</li>
`;

exports[`EcDropdown :props should get disabled when disabled prop is set 1`] = `
<div
  class="ec-dropdown-search ec-dropdown"
  data-test="ec-dropdown"
>
  <div
    class="ec-dropdown-search__trigger"
    data-popover-stub=""
    data-test="ec-popover-dropdown-search"
    disabled="disabled"
    offset="8"
    placement="bottom"
    popoverinnerclass="ec-dropdown-search__popover"
    popperoptions="[object Object]"
  >
    <div
      class="ec-input-field"
      data-test="ec-input-field"
    >
      <!---->
       
      <input
        class="ec-input-field__input ec-input-field__input--has-icon"
        data-test="ec-dropdown__input ec-input-field__input"
        disabled="disabled"
        id="ec-input-field-108"
        placeholder=""
        readonly="readonly"
        type="text"
      />
       
      <div
        class="ec-input-field__icon-wrapper ec-input-field__icon-wrapper--is-disabled"
        data-test="ec-input-field__icon-wrapper"
      >
        <svg
          class="ec-input-field__icon ec-icon"
          data-test="ec-input-field__icon"
          height="24"
          width="24"
        >
          <use
            xlink:href="#ec-simple-arrow-drop-down"
          />
        </svg>
      </div>
       
      <!---->
    </div>
     
    <div>
      <ul
        class="ec-dropdown-search__item-list"
        data-test="ec-dropdown-search__item-list"
      >
        <!---->
         
        <!---->
         
      </ul>
    </div>
  </div>
</div>
`;

exports[`EcDropdown :props should not show the items when the loading is set to true 1`] = `
<div
  class="ec-dropdown-search ec-dropdown"
  data-test="ec-dropdown"
>
  <div
    class="ec-dropdown-search__trigger"
    data-popover-stub=""
    data-test="ec-popover-dropdown-search"
    offset="8"
    placement="bottom"
    popoverinnerclass="ec-dropdown-search__popover"
    popperoptions="[object Object]"
  >
    <div
      class="ec-input-field"
      data-test="ec-input-field"
    >
      <!---->
       
      <input
        class="ec-input-field__input ec-input-field__input--has-icon"
        data-test="ec-dropdown__input ec-input-field__input"
        id="ec-input-field-114"
        placeholder=""
        readonly="readonly"
        type="text"
      />
       
      <div
        class="ec-input-field__icon-wrapper"
        data-test="ec-input-field__icon-wrapper"
      >
        <svg
          class="ec-input-field__icon ec-icon"
          data-test="ec-input-field__icon"
          height="24"
          width="24"
        >
          <use
            xlink:href="#ec-simple-arrow-drop-down"
          />
        </svg>
      </div>
       
      <!---->
    </div>
     
    <div>
      <ul
        class="ec-dropdown-search__item-list"
        data-test="ec-dropdown-search__item-list"
      >
        <!---->
         
        <!---->
         
        <li>
          <div
            class="ec-loading"
            data-test="ec-loading"
          >
            <div
              class="ec-loading__backdrop ec-loading__backdrop--is-transparent"
              data-test="ec-loading__backdrop"
            >
              <svg
                class="ec-loading__icon ec-icon"
                data-test="ec-loading__icon"
                height="24"
                width="24"
              >
                <use
                  xlink:href="#ec-simple-loading"
                />
              </svg>
            </div>
             
            <div
              class="ec-loading__content ec-loading__content--is-transparent"
              data-test="ec-loading__content"
            >
              <div
                class="ec-dropdown-search__loading"
                data-test="ec-dropdown-search__loading"
              />
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`EcDropdown :props should pass errorMessage prop to the triggering input 1`] = `undefined`;

exports[`EcDropdown :props should pass label prop to the triggering input 1`] = `
<label
  class="ec-input-field__label"
  data-test="ec-input-field__label"
  for="ec-input-field-70"
>
  <span
    class="ec-input-field__label-text"
    data-test="ec-input-field__label-text"
  >
    Random label
    <!---->
  </span>
   
  <!---->
</label>
`;

exports[`EcDropdown :props should use placeholder prop 1`] = `
<input
  class="ec-input-field__input ec-input-field__input--has-icon"
  data-test="ec-dropdown__input ec-input-field__input"
  id="ec-input-field-95"
  placeholder="Random placeholder"
  readonly="readonly"
  type="text"
/>
`;

exports[`EcDropdown :props should use searchPlaceholder prop if search is enabled 1`] = `
<li
  class="ec-dropdown-search__search-area"
  data-test="ec-dropdown-search__search-area"
>
  <svg
    class="ec-dropdown-search__search-icon ec-icon"
  >
    <use
      xlink:href="#ec-simple-search"
    />
  </svg>
   
  <input
    autocomplete="off"
    class="ec-dropdown-search__search-input"
    data-test="ec-dropdown-search__search-input"
    placeholder="Random search placeholder"
  />
</li>
`;

exports[`EcDropdown should render as expected 1`] = `
<div
  class="ec-dropdown-search ec-dropdown"
  data-test="ec-dropdown"
>
  <div
    class="ec-dropdown-search__trigger"
    data-popover-stub=""
    data-test="ec-popover-dropdown-search"
    offset="8"
    placement="bottom"
    popoverinnerclass="ec-dropdown-search__popover"
    popperoptions="[object Object]"
  >
    <div
      class="ec-input-field"
      data-test="ec-input-field"
    >
      <!---->
       
      <input
        class="ec-input-field__input ec-input-field__input--has-icon"
        data-test="ec-dropdown__input ec-input-field__input"
        id="ec-input-field-4"
        placeholder=""
        readonly="readonly"
        type="text"
      />
       
      <div
        class="ec-input-field__icon-wrapper"
        data-test="ec-input-field__icon-wrapper"
      >
        <svg
          class="ec-input-field__icon ec-icon"
          data-test="ec-input-field__icon"
          height="24"
          width="24"
        >
          <use
            xlink:href="#ec-simple-arrow-drop-down"
          />
        </svg>
      </div>
       
      <!---->
    </div>
     
    <div>
      <ul
        class="ec-dropdown-search__item-list"
        data-test="ec-dropdown-search__item-list"
      >
        <!---->
         
        <!---->
         
      </ul>
    </div>
  </div>
</div>
`;

exports[`EcDropdown should render readonly input as a trigger 1`] = `
<input
  class="ec-input-field__input ec-input-field__input--has-icon"
  data-test="ec-dropdown__input ec-input-field__input"
  id="ec-input-field-10"
  placeholder=""
  readonly="readonly"
  type="text"
/>
`;
